ci/docker/runtime_functions.sh

```diff
@@ -521,12 +523,30 @@ test_ubuntu_cpu_python3() {
 deploy_docs() {
     set -ex
     pushd .
-    
+
     make docs
 
     popd
 }
 
+deploy_jl_docs {
+    set -ex
+    export PATH="/work/julia/bin:$PATH"
+    export MXNET_HOME='/work/mxnet'
+    export JULIA_PKGDIR='/work/julia-pkg'
+    export DEPDIR=`julia -E 'Pkg.dir()'`
+
+    julia -e 'versioninfo()'
+    ln -sf ${MXNET_HOME}/julia ${DEPDIR}/MXNet
+    # use the prebuilt binary from $MXNET_HOME/lib
+    julia -e 'Pkg.build("MXNet")'
+    # build docs
+    julia -e 'Pkg.add("Documenter")'
+    julia -e 'cd(Pkg.dir("MXNet")); include(joinpath("docs", "make.jl"))'
+    # TODO: make Jenkins worker push to MXNet.jl ph-pages branch if master build
```

Comment:
You said you needed help here. How may I help you?

Reply:
There is a builtin deployment method in Documenter:
https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#SSH-Deploy-Keys-1
It needs a key pair: add a public key on GitHub (only @pluskid has the permission) first, then 
add an env var `DOCUMENTER_KEY` as private key in Jenkins.

Or... will it make deployment easier, if it outputs docs to a specific dir and let Jenkins do the pushing work? I'm not sure how to make Jenkins gain the permission of push to gh-pages.

Reply:
I see, we're basically using the same approach for our MXNet docs. I have to say that the name of this job is a bit off, because we're not deploying to production as part of this pipeline.

We have a separate publish job for our website. It gets two env-vars passed into the job and uses them as follows:
```
git clone https://$APACHE_USERNAME:$APACHE_PASSWORD@github.com/apache/incubator-mxnet-site
```
I think we could follow the same approach here, right? 

About the private key: Usually you use a GitHub personal access token. The username is the GitHub-Accountname and the password is the generated access token. Is this suitable here? I'm happy to assist you with creating the jenkins job.

Reply:
> because we're not deploying to production as part of this pipeline

Oh, is it ok to put this publish job here?

> I think we could follow the same approach here, right?

Which GitHub bot account have you set up already? 
@pluskid: Could you add it as collaborator?

Reply:
The publish job here is just for staging - in order to verify document generation as part of the PR. The publish for the main website will be done with a different job. You might want to work together with @aaronmarkham on this one.

We're not allowed to use bot accounts for Apache services, so right now, we're using a committers account credentials.. Shall I set one up for this purpose?

Reply:
I think using committer's account credentials isn't a good idea.
Maybe deploy key is a better way: https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys

Reply:
That sounds feasible :)